### PR TITLE
Fix godoc wording and inconsistencies

### PR DESCRIPTION
**Pull request by rakyll**
1. Move the component lifecycle to interface godoc so the godoc can render it
2. Avoid using below/above, godoc already groups things correctly.
3. Avoid giving references to what functions/methods implement what. Instead, mention what they do.
4. Fix inconsistencies in new lines for long lines.

Related to #3049.

### CODE REVIEW

File: component/receiver.go [hunk 1]
```diff
@@ -23,37 +23,39 @@ import (
 	"go.opentelemetry.io/collector/consumer"
 )
 
-// Receiver defines functions that trace and metric receivers must implement.
+// Receiver allows the collector to recieve metrics, traces and logs.
```
Comment (tigrannajaryan) [inline component/receiver.go:26]:
```suggestion
// Receiver allows the collector to receive metrics, traces and logs.
```

File: component/receiver.go [hunk 2]
```diff
@@ -23,37 +23,39 @@ import (
...
 type Receiver interface {
 	Component
 }
 
-// A TracesReceiver is an "arbitrary data"-to-"internal format" converter.
-// Its purpose is to translate data from the wild into internal trace format.
+// A TracesReceiver recieves traces.
```
Comment (tigrannajaryan) [inline component/receiver.go:31]:
```suggestion
// A TracesReceiver receives traces.
```

File: component/receiver.go [hunk 3]
```diff
@@ -23,37 +23,39 @@ import (
...
 // TracesReceiver feeds a consumer.Traces with data.
 //
 // For example it could be Zipkin data source which translates Zipkin spans into pdata.Traces.
 type TracesReceiver interface {
 	Receiver
 }
 
-// A MetricsReceiver is an "arbitrary data"-to-"internal format" converter.
-// Its purpose is to translate data from the wild into internal metrics format.
+// A MetricsReceiver recieves metrics.
```
Comment (tigrannajaryan) [inline component/receiver.go:40]:
```suggestion
// A MetricsReceiver receives metrics.
```

File: component/receiver.go [hunk 4]
```diff
@@ -23,37 +23,39 @@ import (
...
 // MetricsReceiver feeds a consumer.Metrics with data.
 //
 // For example it could be Prometheus data source which translates Prometheus metrics into pdata.Metrics.
 type MetricsReceiver interface {
 	Receiver
 }
 
-// A LogsReceiver is a "log data"-to-"internal format" converter.
-// Its purpose is to translate data from the wild into internal data format.
+// A LogsReceiver recieves logs.
```
Comment (tigrannajaryan) [inline component/receiver.go:49]:
```suggestion
// A LogsReceiver receives logs.
```

**Comment by rakyll** [comment]
PTAL.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**